source/env_tutorial/bsuite.rst

```diff
@@ -0,0 +1,190 @@
+
+.. code:: shell
+
+   # Method1: Install Directly
+   pip install bsuite
+   # Method2: Install with DI-engine requirements
+   cd DI-engine
+   pip install ".[common_env]"
+Verify installation
+--------
+
+Once installed, you can verify whether the installation is successful by running the following command on the Python command line.
+
+.. code:: python
+
+   import bsuite
+   env = bsuite.load_from_id('catch/0')
+   timestep = env.reset()
+   print(timestep)
+Original environment space
+========================
+
+.. _Observations space-1:
+Observations space
+--------
+
+-  Array representing the state of the environment, dimensions and size can vary according to the specific environment, datatype is \ ``np.float32``.
+
+.. _Actions space-1:
+Actions space
+--------
+
+-  The action space is a discrete space of size N, (where N can vary according to the environment), datatype is \ ``int``\ and input is a python integer value（or a np array of dimension 0 such as \ ``np.array(1)``\ to input action 1）.
+
+-  For example, in the Deep Sea environment, N is equal to 2, thus action values ranges from 0 to 1. For their specific meaning you can refer to the following list.
+
+   -  0：LEFT.
+
+   -  1：RIGHT.
+
+.. _Rewards space-1:
+Rewards space
+--------
+
+-  Rewards are assigned according to the rules of the environments, rewards are usually a \ ``float``\ value.
+
+.. _Other-1:
+Others
+----
+
+-  Environments terminate once they have reached their maximum number of steps or is encountered a failure state. All environments have a fixed number of maximum steps, but not all environments have a failure state.
+
+Key facts
+========
+
+1. Each environment contains several configurations to make it gradually more challenging.
+
+2. Discrete actions space.
+
+3. Each environment is designed to test a particular propriety of RL policies, including: generalization, exploration, credit assignment, scaling, noise, memory.
+
+4. Rewards scale can vary a lot.
+
+.. _Other-3:
+Others
+====
+
+Using bsuite in 'OpenAI Gym' format
+----------
+
+Our implementation uses the bsuite Gym wrapper to make the bsuite codebase run under the OpenAI Gym interface. Hence, ``gym`` needs to be installed to make bsuite work properly.
+
+.. code:: python
+
+   import bsuite
+   from bsuite.utils import gym_wrapper
+   env = bsuite.load_and_record_to_csv('catch/0', results_dir='/path/to/results')
+   gym_env = gym_wrapper.GymFromDMEnv(env)
+
+About configurations
+--------
+
+Configurations are designed to increase the level of difficulty of an environment. For example, in a 5-armed bandit environment, configurations are used to regulate the level of noise to perturb the rewards.
+Given a specific environment, all possible configurations can be visualized with the following code snippet.
+
+.. code:: python
+
+   from bsuite import sweep
+   print('List bsuite_id for "bandit_noise" experiment:')
+   print(sweep.BANDIT_NOISE)
```

Comment:
I believe the should be generated through the following code snippet:
from bsuite import sweep
for bsuite_id in sweep.BANDIT_NOISE:
  env = bsuite.load_from_id(bsuite_id)
  print('bsuite_id={}, settings={}, num_episodes={}'
        .format(bsuite_id, sweep.SETTINGS[bsuite_id], env.bsuite_num_episodes))

Reply:
Yes, this code provides more information